device/pairing.go

```diff
@@ -50,11 +50,12 @@ func (d *Device) obtainNewCertificate() error {
 	return d.saveCertificateFromString(cert)
 }
 
-func (d *Device) getBrokerURL() (string, error) {
+func (d *Device) getBrokerURL() error {
```

Comment:
Since this doesn't return the broker URL anymore but it saves it inside the struct, change the name to `ensureBrokerURL`.